libs/portals/core/src/utils/modules.ts

```diff
@@ -82,3 +82,31 @@ export const arrangeRoutes = async ({
 
   return filteredRoutes
 }
+
+type GetActiveModuleArgs = {
+  userInfo: User
+  modules: PortalModule[]
+  apolloClient: ApolloClient<NormalizedCacheObject>
+  pathname: string
+}
+
+export const getActiveModule = ({
+  pathname,
+  userInfo,
+  modules,
+  apolloClient,
+}: GetActiveModuleArgs) =>
+  modules.find((module) =>
+    module
+      // Get all routes for the module
+      .routes({
+        userInfo,
+        client: apolloClient,
```

Comment:
This `client` argument was added to support async dynamic routes, but it is not used currently and the types ban it. Only in the `global` function returns a Promise.

Should get rid of the GraphQL client until we need it again.